core/devops.go

```diff
@@ -171,7 +171,14 @@ func (d *Devops) invokeOrQuery(ctx context.Context, chaincodeInvocationSpec *pb.
 	}
 
 	// Now create the Transactions message and send to Peer.
-	uuid := util.GenerateUUID()
+        var userGivenID = chaincodeInvocationSpec.UserGivenID
+        var uuid string
+        if userGivenID != "" {
+                uuid = userGivenID
```

Comment:
@muralisrini I think that is not needed.That would imply enormous performance problems (e.g. if one has a big ledger it would need an expensive lookup).